{
  "comments": [
    {
      "key": {
        "uuid": "f77cb380_fcb5cf85",
        "filename": "logcat/logcat.cpp",
        "patchSetId": 1
      },
      "lineNbr": 1837,
      "author": {
        "id": 1032276
      },
      "writtenOn": "2017-10-18T23:42:39Z",
      "side": 0,
      "message": "man close pedantically:\n\nRETURN VALUE\n       close()  returns  zero on success.  On error, -1 is returned, and errno\n       is set appropriately.\n\nERRORS\n       EBADF  fd isn\u0027t a valid open file descriptor.\n\n       EINTR  The close() call was interrupted by a signal; see signal(7).\n\n       EIO    An I/O error occurred.\n\nalthough this is a non-starter in Android ...",
      "range": {
        "startLine": 1837,
        "startChar": 0,
        "endLine": 1837,
        "endChar": 31
      },
      "revId": "4ff2607ef719bade2b45538878136a7d71621882",
      "serverId": "85c56323-6fa9-3386-8a01-6480fb634889",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "cebe06e0_6e802d74",
        "filename": "logcat/logcat.cpp",
        "patchSetId": 1
      },
      "lineNbr": 1837,
      "author": {
        "id": 1079148
      },
      "writtenOn": "2017-10-18T23:56:24Z",
      "side": 0,
      "message": "Saving and restoring errno for just this one close call seems pointless when there\u0027s another close and two function calls that call close right above it, and probably another one below.",
      "parentUuid": "f77cb380_fcb5cf85",
      "range": {
        "startLine": 1837,
        "startChar": 0,
        "endLine": 1837,
        "endChar": 31
      },
      "revId": "4ff2607ef719bade2b45538878136a7d71621882",
      "serverId": "85c56323-6fa9-3386-8a01-6480fb634889",
      "unresolved": false
    }
  ]
}